app/models/mixins/custom_actions_mixin.rb

```diff
@@ -25,19 +25,22 @@ def custom_action_buttons(applies_to = self)
   end
 
   def generic_button_group
-    generic_custom_buttons.select { |button| !button.parent.nil? }
+    generic_custom_buttons.includes(:custom_button_sets)
+                          .select { |button| button.custom_button_sets.present? }
   end
 
   def custom_button_sets_with_generics
-    custom_button_sets + generic_button_group.map(&:parent).uniq.flatten
+    custom_button_sets + generic_button_group.map(&:custom_button_sets).flatten.uniq
   end
 
   def custom_buttons
-    generic_custom_buttons.select { |button| button.parent.nil? } + direct_custom_buttons
+    generic_custom_buttons.includes(:custom_button_sets)
+                          .select { |b| b.custom_button_sets.blank? } + direct_custom_buttons
```

Comment:
`button.custom_button_sets` is a `has_many` relation define with this:

```ruby
    has_many miq_set_class.name.underscore.pluralize.to_sym,
             :class_name => miq_set_class.to_s, # rubocop:disable Rails/ReflectionClassName
             :source     => miq_set_class.name.underscore.to_sym,
             :through    => :miq_set_memberships
```

Where `.parent` is a `has_one`, so `.nil?` doesn't work like it used to because it would return `[]`.  This is why it was switched to use `.blank?`.

The reason we used `.present?` for the one above, because it kept the logic very similar and it would "`.reject`" any `[]` values as it would have done with `.nil` before.

I will change the one below to `.blank?` for consistency though, just so we are using the inverse of `.present?`.